io/src/lib.rs

```diff
@@ -50,13 +52,46 @@ pub trait Read {
 
     #[inline]
     fn take(&mut self, limit: u64) -> Take<Self> { Take { reader: self, remaining: limit } }
+
+    /// Attempts to read up to limit bytes from the reader, allocating space in `buf` as needed.
+    ///
+    /// `limit` is used to prevent a denial of service attack vector since an unbounded reader will
+    /// exhaust all memory.
+    ///
+    /// Similar to `std::io::Read::read_to_end` but with the DOS protection.
+    #[cfg(any(feature = "alloc", feature = "std"))]
+    #[inline]
+    fn read_to_limit(&mut self, buf: &mut Vec<u8>, limit: u64) -> Result<usize> {
```

Comment:
I think `limit` should be `usize` so 32-bit platforms don't shoot themselves in the foot.

And we should probably use `alias`.

Also did you check whether rustc suggests this if you type `read_to_end`?

Reply:
> I think `limit` should be `usize` so 32-bit platforms don't shoot themselves in the foot.

I don't understand what is your final position on this, if it should be `usize` then we should change `take` and I did that before and its worse? FWIW `std` uses a `u64` for take.

> And we should probably use `alias`.

I don't understand this comment?

> Also did you check whether rustc suggests this if you type `read_to_end`?

I don't think we are going to get anything nice out of the compiler, currently its:
```bash
175 | pub struct Cursor<T> {
    | -------------------- method `read_to_end` not found for this struct
...
340 |         let read = reader.read_to_end(&mut buf, 32).expect("failed to read to limit");
    |                           ^^^^^^^^^^^ method not found in `Cursor<&String>`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `read_to_end`, perhaps you need to implement it:
            candidate #1: `std::io::Read`
```

Reply:
My final position is that it doesn't matter because a constant is used for DoS protection, so let's do the easy thing and not change the code.

> I don't understand this comment 

https://doc.rust-lang.org/rustdoc/advanced-features.html#add-aliases-for-an-item-in-documentation-search

Sad about rustc error but I suspect it's not solvable.

Reply:
Oh nice doc trick, I didn't know that one. Will add.

Reply:
lol, and I claimed the other day to the kids that my memory works really well, just only when I'm programming.